language/python/develop.md

```diff
@@ -4,8 +4,253 @@ keywords: python, local, development, run,
+  -e MYSQL_ALLOW_EMPTY_PASSWORD=true \
+  mysql
+```
+
+Now, let’s make sure that our MySQL database is running and that we can connect to it. Connect to the running MySQL database inside the container using the following command:
+
+```shell
+$ docker run -it --network mysqlnet --rm mysql mysql -hmysqldb
+Enter password: ********
+
+Welcome to the MySQL monitor.  Commands end with ; or \g.
+Your MySQL connection id is 8
+Server version: 8.0.23 MySQL Community Server - GPL
+
+Copyright (c) 2000, 2021, Oracle and/or its affiliates.
+
+Oracle is a registered trademark of Oracle Corporation and/or its
+affiliates. Other names may be trademarks of their respective
+owners.
+
+Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
+
+mysql>
+```
+
+In the above command, we used the same MySQL image to connect to the database but this time, we passed the ‘mysql’ command to the container with the `-h` flag containing the name of our MySQL container name. Press CTRL-D to exit the MySQL  interactive terminal.
+
+Okay, now that we have a running MySQL, let’s update `app.py` to use MySQL as a datastore. Let’s also add some routes to our server. One for fetching records and one for inserting records.
+
+```shell
+import mysql.connector
+import json
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
+  return 'Hello, Docker!'
+
+@app.route('/widgets')
+def get_widgets() :
+  mydb = mysql.connector.connect(
+    host="mysqldb",
+    user="root",
+    password="p@ssw0rd1",
+    database="inventory"
+  )
+  cursor = mydb.cursor()
+
+
+  cursor.execute("SELECT * FROM widgets")
+
+  row_headers=[x[0] for x in cursor.description] #this will extract row headers
+
+  results = cursor.fetchall()
+  json_data=[]
+  for result in results:
+    json_data.append(dict(zip(row_headers,result)))
+
+  cursor.close()
+
+  return json.dumps(json_data)
+
+@app.route('/db')
+def db_init():
+  mydb = mysql.connector.connect(
+    host="mysqldb",
+    user="root",
+    password="p@ssw0rd1"
+  )
+  cursor = mydb.cursor()
+
+  cursor.execute("DROP DATABASE IF EXISTS inventory")
+  cursor.execute("CREATE DATABASE inventory")
+  cursor.close()
+
+  mydb = mysql.connector.connect(
+    host="mysqldb",
+    user="root",
+    password="p@ssw0rd1",
+    database="inventory"
+  )
+  cursor = mydb.cursor()
+
+  cursor.execute("DROP TABLE IF EXISTS widgets")
+  cursor.execute("CREATE TABLE widgets (name VARCHAR(255), description VARCHAR(255))")
+  cursor.close()
+
+  return 'init database'
+
+if __name__ == "__main__":
+  app.run(host ='0.0.0.0')
+```
+
+We’ve added the MySQL module and updated the code to connect to the database server, created a database and table. We also created a couple of routes to save widgets and fetch widgets. We now need to rebuild our image so it contains our changes.
+
+First, let’s add the `mysql-connector-python `module to our application using pip.
+
+```shell
+$ pip install mysql-connector-python
```

Comment:
Maybe it would be easier to manually add the dependency to `requirements.txt`. Same as in https://github.com/docker/awesome-compose/blob/master/nginx-flask-mysql/backend/requirements.txt

From what I understand `pip freeze` command dumps all installed local packages, not only the ones we need for the app. I may be wrong, I have never used `pip freeze` before :sweat_smile:

Reply:
Flask==1.1.1
mysql-connector==2.2.9